s3.tf

```diff
@@ -89,26 +93,43 @@ resource "aws_s3_bucket_inventory" "apiary_bucket" {
 }
 
 resource "aws_s3_bucket_public_access_block" "apiary_bucket" {
-  count  = var.s3_block_public_access == true ? "${length(local.apiary_data_buckets)}" : 0
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.s3_block_public_access == true ? {
```

Comment:
when we added this, discussed possibility of removing the variable and block public access by default, so we should be able remove `s3_block_public_access` variable and above condition as we are creating breaking change.

Reply:
Same question as above - migration script can't handle this for user.  Is this OK?